.github/workflows/test-build.yml

```diff
@@ -0,0 +1,70 @@
+name: Build tests
+on: [pull_request]
+jobs:
+
+  kirkstone-repo:
+    runs-on: self-hosted
+    if: "!contains(github.event.head_commit.message, 'ci skip')"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/bitbake-repo
+        with:
+          repo_release: 'kirkstone'
+
+  matrix-build-image:
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
nice. I was not aware of this 

Doc:https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs